nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,561 @@
+    """ Returns an HTML table with pertinent model attributes & information.
+    Does not contain derived attributes.
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object.
+    
+    Returns
+    -------
+    HTML Table: String
+        HTML code for creating a table.
+    """
+    selected_attributes = [
+        'subject_label',
+        'mask_img',
+        'drift_model',
+        'hrf_model',
+        'standardize',
+        'noise_model',
+        'min_onset',
+        't_r',
+        'labels_',
+        'high_pass',
+        'target_shape',
+        'signal_scaling',
+        'drift_order',
+        'scaling_axis',
+        'smoothing_fwhm',
+        'target_affine',
+        'slice_time_ref',
+        'fir_delays',
+        ]
+    selected_model_attributes_values = {
+        attr_name: model.__dict__[attr_name]
+        for attr_name in selected_attributes
+        if attr_name in model.__dict__
+        }
+    model_attributes_table = pd.DataFrame.from_dict(selected_model_attributes_values,
+                                                    orient='index',
+                                                    )
+    model_attributes_table = model_attributes_table.to_html(header=False,
+                                                            sparsify=False,
+                                                            )
+    return model_attributes_table
+
+
+def make_statistical_maps(model, contrasts):
+    """ Given a model and contrasts, return the corresponding z-maps
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object
+    
+    contrasts: Dict[str, ndarray or str]
+        Dict of contrasts
+    
+    Returns
+    -------
+    statistical_maps: Dict[str, niimg]
+        Dict of statistical maps keyed to contrast names.
+    """
+    statistical_maps = {contrast_id: model.compute_contrast(contrast_val)
+                        for contrast_id, contrast_val in contrasts.items()
+                        }
+    return statistical_maps
+
+
+def _make_html_for_design_matrices(model):
+    """ Accepts a FirstLevelModel or SecondLevelModel object
+    with fitted design matrices & generates HTML code
+    to insert their plots into the report.
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object
+        First or Second Level Model objects with fitted design matrices.
+        
+    Returns
+    -------
+    html_design_matrices: String
+        HTML code for the plotted design matrices,
+        to be inserted into the HTML template.
+    """
+    html_design_matrices = []
+    for count, design_matrix in enumerate(model.design_matrices_):
+        design_matrix_image_axes = plot_design_matrix(design_matrix)
+        buffer = io.StringIO()
+        design_matrix_image_axes.figure.savefig(buffer, format='svg')
+        html_design_matrix = (
+            '<svg class="dmtx">{}</svg>'.format(buffer.getvalue()))
```

Comment:
use an <img> instead of inline svg

Reply:
an... img?

Reply:
sorry, yes

Reply:
- From my readings, Inlined SVGs are expected to have the `<svg>` tag as per the spec; not `img` or `object`.
 - I have removed HTML string manipulation in Python code from here.
 - Matplotlib autom inserts the `svg` tag so we don't have to add it in the template.

Reply:
inlined svg is not what we want. we want to embed an svg image as a data url, as in this example https://gist.github.com/jeromedockes/2114af2838c70748d48846ef1eb379c0

Reply:
Ah the missing link. That's why I couldn't get save image to work. I was missing pieces of the puzzle. Thanks @jeromedockes . Did I mention you are my 14th favourite person?

Reply:
> Ah the missing link

sorry I thought I had posted it earlier

Reply:
You had, I am commenting on it now.